frmts/ers/ersdataset.cpp

```diff
@@ -816,23 +816,20 @@ int ERSDataset::Identify( GDALOpenInfo * poOpenInfo )
 /* -------------------------------------------------------------------- */
 /*      We assume the user selects the .ers file.                       */
 /* -------------------------------------------------------------------- */
-    if( poOpenInfo->nHeaderBytes > 15
-        && STARTS_WITH_CI((const char *) poOpenInfo->pabyHeader, "Algorithm Begin") )
+    CPLString osHeader((const char *)poOpenInfo->pabyHeader, poOpenInfo->nHeaderBytes);
+
+    if( osHeader.ifind( "Algorithm " ) != std::string::npos )
```

Comment:
why removing the " Begin" ?
The check could potentially now trigger more easily on unrelated datasets. I'm not familiar with that format, but if those algorithm ERS files have also a "DatasetHeader " string in them, perhaps this check could be moved in the below check ? Otherwise we may want to add additional conditions

Reply:
Agreed, the ` Begin` should stay and I've put it back in here.  This will preserve the existing behavior as much as possible, while still allowing comments to occur even at the start of algorithm files.